### ISSUE TITLE
Frequent crashes

### ISSUE THREAD

**Issue by sossalemaire** [home-sweet-gnome/dash-to-panel#597]
First of all, thanks for a great extension, it's exactly what I needed.

Unfortunalety, my gnome shell crashes frequently when using it (around two or three times a day), and from what I see in journalctl /usr/bin/gnome-shell gives, it seems to come from this extension:
`mars 11 12:52:19 SossaTurbo gnome-shell[2209]: Source ID 153 was not found when attempting to remove it`
`mars 11 12:52:20 SossaTurbo gnome-shell[2209]: JS LOG: Extension "dash-to-panel@jderose9.github.com" had error: Error: Type name Gjs_DashToPanel-TaskbarActor is already registered`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: st_theme_context_get_for_stage: assertion 'CLUTTER_IS_STAGE (stage)' failed`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: g_object_get: assertion 'G_IS_OBJECT (object)' failed`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: st_theme_context_get_for_stage: assertion 'CLUTTER_IS_STAGE (stage)' failed`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: g_object_get: assertion 'G_IS_OBJECT (object)' failed`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: st_theme_context_get_for_stage: assertion 'CLUTTER_IS_STAGE (stage)' failed`
`mars 11 12:52:21 SossaTurbo gnome-shell[2209]: g_object_get: assertion 'G_IS_OBJECT (object)' failed`
Please let me know if you need anything else to troubleshoot.

**Comment by charlesg99** [comment]
Hi sossalemaire, thank you for the report. How did you install dash-to-panel and which version is it? What version of gnome-shell are you on? Is the log `Error: Type name Gjs_DashToPanel-TaskbarActor is already registered on your system` happening right before the crash? This error is weird as it looks like you have 2 installations of dash-to-panel loading when gnome-shell is initializing.

Also, could you please try the latest[ development version from github ](https://github.com/home-sweet-gnome/dash-to-panel/wiki/Installation#installation-from-github)and see if this helps? Thanks!

**Comment by sossalemaire** [comment]
I installed dash-to-panel from extensions.gnome.org. My version of gnome-shell is 3.22.3 (I'm using an up-to-date Debian 9). I don't know if it happens before or after the crash, but it happens almost at the same time. I've not been able to detect a triggering event, it seems random. I'll try the development version and I'll let you know.

**Comment by charlesg99** [comment]
The current version on e.g.o is v17 and this version had a bug that caused a crash when opening more than 4 windows of the same application (see #552, #573 and #580). Could it be the cause of your crash as well? Thanks!

**Comment by sossalemaire** [comment]
I installed the latest version from github, I have the same crash problem. As far as I can tell, I don't have any application with more than 4 windows. The only thing I always see in syslog right before the crash is "Source ID XXX was not found when attempting to remove it" (which seems to be an error logged when an application tries to unregistrer an already unregistered event: https://stackoverflow.com/questions/23199699/glib-critical-source-id-xxx-was-not-found-when-attempting-to-remove-it) . Then I see a bunch of "Gs  no app for changed [extension name]" for all installed extension (many of which are not enabled), and then I see the "Type name Gjs_DashToPanel-TaskbarActor is already registered" error. It seems like dash-to-panel tries to register itself again; maybe gnome does some kind of extension probe, dash-to-panel tries to unregister things, fails to do so, then registers itself again because it thinks it's unregistered, and boom.

**Comment by charlesg99** [comment]
Yeah the crash due to 5 windows of the same app is fixed on the github version so it can't be that then. Next time it crashes, afterwards please run `journalctl --since='20 minutes ago'` and send us the output so we can have a look. Thanks!

**Comment by sossalemaire** [comment]
Another crash, here is the log, the error is at 16:52:37.
[journal.log](https://github.com/home-sweet-gnome/dash-to-panel/files/2967365/journal.log)

**Comment by charlesg99** [comment]
Thank you for the log, although I don't think this warning is the source of the problem and I don't see a typical gnome-shell crash in there either. When you say crash, what happens exactly? Gnome-shell restarts? You go back to the login screen? Also, what are you doing before the crash? I do have an installation of Debian 9 with dash to panel so I could try to reproduce it. What other extensions are enabled on your machine? Thanks!

**Comment by sossalemaire** [comment]
My symptom is: my screen becomes unresponsive for around ten seconds, then the dash-to-panel panel disappears. I don't go back to the login screen, the x11 server is still up (I use x11 instead of wayland because with x11 I can restart gnome-shell to get dash-to-panel back :)). The other extensions are still up too.
The latest crash for which I sent the log happened while I was preparing tea, I was not doing anything on the computer... It doesn't seem to be triggered by something I do.
The following extensions are enabled:
- Applications menu: https://extensions.gnome.org/extension/6/applications-menu/
- BackSlide: https://extensions.gnome.org/extension/543/backslide/
- dash-to-panel, obviously :)
I use a laptop with a dock and two screens most of the time, but the "crash" happens even when I only use the laptop's integrated screen. Thanks for helping me, this problem seems to be quite peculiar!

**Comment by mweirauch** [comment]
Well, the second comment on the Backslide extension page mentions slow downs and freezes. Perhaps worth a try living without it for a while and seeing if the crashes still persist?

**Comment by charlesg99** [comment]
I did some tests and the cause is definitely Backslide. I modified the delay between background image changes to 1 seconds and I can reproduce the issue pretty consistently within 1-2 minutes or so. I didn't fully investigate why gnome-shell reloads all extensions without crashing though (same happens on Wayland), but when it does, extensions aren't correctly disabled, hence the dash to panel type error.

Will wait for sossalemaire to confirm that it doesn't happen when Backslide is disabled. Thanks!
